AddOns/BasicFinancials/app/src/codeunit/ExperienceTier.Codeunit.al

```diff
@@ -0,0 +1,58 @@
+codeunit 20602 "Experience Tier BF"
+{
+    Access = Internal;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Application Area Mgmt. Facade", 'OnValidateApplicationAreas', '', true, true)]
+    local procedure OnValidateApplicationAreas(ExperienceTierSetup: Record "Experience Tier Setup"; TempApplicationAreaSetup: Record "Application Area Setup")
+    var
+    begin
+        if not ExperienceTierSetup."BF Basic Financials" then
+            exit;
+
+        TempApplicationAreaSetup.TestField("BF Orders", false);
+        TempApplicationAreaSetup.TestField("BF Basic Financials", true);
+    end;
+
+    [EventSubscriber(ObjectType::Codeunit, Codeunit::"Application Area Mgmt. Facade", 'OnSetExperienceTier', '', true, true)]
+    local procedure SetApplicationAreas(ExperienceTierSetup: Record "Experience Tier Setup"; var ApplicationAreasSet: Boolean; var TempApplicationAreaSetup: Record "Application Area Setup")
+    var
+        AppAreaMgmt: Codeunit "App Area Mgmt BF";
+    begin
+        if not ExperienceTierSetup."BF Basic Financials" then
+            exit;
+
+        AppAreaMgmt.GetEssentialExperienceAppAreas(TempApplicationAreaSetup);
+        DisableExperienceAppAreas(TempApplicationAreaSetup);
+        GetBasicFinancialsExperienceAppAreas(TempApplicationAreaSetup);
+        ApplicationAreasSet := true;
+    end;
+
+    local procedure GetBasicFinancialsExperienceAppAreas(var TempApplicationAreaSetup: Record "Application Area Setup" temporary)
+    var
+    begin
+        TempApplicationAreaSetup."BF Orders" := false;
+        TempApplicationAreaSetup."BF Basic Financials" := true;
+    end;
+
+    local procedure DisableExperienceAppAreas(var TempApplicationAreaSetup: Record "Application Area Setup" temporary)
+    var
+    begin
+        TempApplicationAreaSetup."Relationship Mgmt" := false;
+        TempApplicationAreaSetup.Assembly := false;
+        TempApplicationAreaSetup."Item Charges" := false;
+        TempApplicationAreaSetup.Intercompany := false;
+        TempApplicationAreaSetup."Sales Return Order" := false;
+        TempApplicationAreaSetup."Purch Return Order" := false;
+        TempApplicationAreaSetup."Cost Accounting" := false;
+        TempApplicationAreaSetup."Sales Budget" := false;
+        TempApplicationAreaSetup."Purchase Budget" := false;
+        TempApplicationAreaSetup."Item Budget" := false;
+        TempApplicationAreaSetup."Sales Analysis" := false;
+        TempApplicationAreaSetup."Purchase Analysis" := false;
+        TempApplicationAreaSetup."Inventory Analysis" := false;
+        TempApplicationAreaSetup."Item Tracking" := false;
+        TempApplicationAreaSetup."Order Promising" := false;
+        TempApplicationAreaSetup.Reservation := false;
+        TempApplicationAreaSetup.ADCS := false;
```

Comment:
And if we add more Application Areas? Wouldn't be better to take the other approach and enable only Application Areas we need?

Reply:
We have previously discussed this, and I think the conclusion was that if Microsoft add more Application Areas it will probably be in the Basic Financial license. And therefore is the current approach choosed.
I think it is up to Microsoft to decide which approach, you want me to implement. Please reconcile it internally in Microsoft and then notify if you want to change approach.